src/pages/settings/panes/Appearance.tsx

```diff
@@ -446,11 +452,15 @@ export const Appearance = connectState(Component, (state) => {
     };
 });
 
-function getContrastingColour(hex: string){
-    hex = hex.replace("#", "");
-    const r = parseInt(hex.substr(0,2),16);
-    const g = parseInt(hex.substr(2,2),16);
-    const b = parseInt(hex.substr(4,2),16);
-    const cc = ((r*299)+(g*587)+(b*114))/1000;
-    return (cc >= 128) ? 'black' : 'white';
+function getContrastingColour(hex: string, fallback: string): string {
+    try {
+        hex = hex.replace("#", "");
+        const r = parseInt(hex.substr(0, 2), 16);
+        const g = parseInt(hex.substr(2, 2), 16);
+        const b = parseInt(hex.substr(4, 2), 16);
+        const cc = (r * 299 + g * 587 + b * 114) / 1000;
+        return cc >= 175 ? "black" : "white";
+    } catch (e) {
+        return getContrastingColour(fallback, "#fffff");
+    }
```

Comment:
Try catch won't work in this situation since JavaScript does not throw errors on failing to pares.

Instead of the try catch block, needs something like this:

```javascript
const b = parseInt(hex.substr(4, 2), 16);
if (isNaN(a) || isNaN(b) || isNaN(c)) return getContrastingColour(fallback, "#fffff");

const cc = (r * 299 + g * 587 + b * 114) / 1000;
```